### PR TITLE
Added flag to choose interface to discover on

### PR DESCRIPTION
**Pull request by eoswald**

### CODE REVIEW

File: autoroute/main.go [hunk 1]
```diff
@@ -130,8 +118,32 @@ func main() {
...
+		public_key := key.PublicKey()
+		if *dev_name == "" {
+			log.Print("Started probing of all interfaces")
+
+			devs, err := net.Interfaces()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			go func() {
```
Comment (c00w) [inline autoroute/main.go]: Can you unify these code branches and just populate a list of interfaces?

File: autoroute/main.go [hunk 2]
```diff
@@ -47,13 +56,16 @@ func GetLinkLocalAddr(dev net.Interface) (*net.IPAddr, error) {
 	}
 
 	if ll_addr == nil {
-		log.Fatalf("Couldn't find interface link-local addresses %v", dev.Name)
+		msg := fmt.Sprintf("Couldn't find link-local address on interface %v", dev.Name)
+		e := LinkLocalError{msg, false}
```
Comment (c00w) [inline autoroute/main.go:60]: Why isn't this fatal? Don't we need a link local address?

Comment (eoswald) [inline autoroute/main.go:60]: I guess it should be fatal if _none_ of the interfaces have link locals. But if I only have a wireless connection of course I'm not going to get a link local on my ethernet device.

File: autoroute/main.go [hunk 3]
```diff
@@ -130,12 +144,30 @@ func main() {
 	n := node.NewServer(key)
 
 	if *autodiscover {
-		log.Print("Starting Probing of all interfaces")
-		go Probe(key, n)
+		var devs []net.Interface
```
Comment (c00w) [inline autoroute/main.go]: I think you need to do devs := make(net.Interface, 0)

**Pull request merged**